### PR TITLE
fix!: output-module-groups to work like run-all

### LINKED ISSUE #2685: `terragrunt output-module-groups` fails if not run from a Terragrunt module
**Describe the bug**
The description of `output-module-groups` says:

> This will recursively search the current working directory for any folders that contain Terragrunt modules and build the dependency graph based on dependency and dependencies blocks.

However, if I run `terragrunt output-module-groups` from the root of my repo, the command fails. I do not have a terragrunt.hcl at root, as all Terragrunt modules are organized within my repo. Here's an example of my layout:

```
- root
  - aws
    - aws.hcl (shared config for all AWS Terragrunt modules)
    - commercial
      - account1
        - terragrunt.hcl
        - test.tfvars
      - account2
        - terragrunt.hcl
        - test.tfvars
        - submodule
          - terragrunt.hcl
          - test.tfvars
  - azure
    - azure.hcl (shared config for all Azure Terragrunt modules)
    - ...
```

Here's the error I'm getting:
```
$ terragrunt output-module-groups
ERRO[0000] Error reading file at path /terragrunt_test/terragrunt.hcl: open /terragrunt_test/terragrunt.hcl: no such file or directory
ERRO[0000] Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```

**To Reproduce**
Attempt to run `terragrunt output-module-groups` from a a directory that contains Terragrunt modules in subfolders, but is not a Terragrunt module itself.

**Expected behavior**
I would expect this command to recursively search the current directory for all Terragrunt modules and build the dependency graph, even if there is no terragrunt.hcl in the current dir.

**Workaround**
As a workaround, you can create an empty terragrunt.hcl in the root directory to get it to recursively search for all Terragrunt modules.

**Versions**
- Terragrunt version: v0.50.6
- Terraform version: v1.5.4
- Environment details (Ubuntu 20.04, Windows 10, etc.): Rocky Linux 8

### LINKED ISSUE #2768: output-module-groups does not generate destroy order or support terragrunt cli options
**Describe the solution you'd like**
output-module-groups does not generate the proper destroy order if passed in the destroy command

it also does not support options like`--terragrunt-modules-that-include`

it would be great if these features were supported so output-module-groups ran exactly like run-all in terms of forming a stack.

**Describe alternatives you've considered**
Writing a new command to handle these features

### PR DESCRIPTION
**Pull request by bt-macole**
## Description

output-module-groups was not working from a directory that did not have a terragrunt.hcl resource and it was not working with terragrunt cli options such as `terragrunt-modules-that-include`.

furthermore, it was not inverting order for the destroy command.

and required init even though it is just using the stackconfig to output planned behavior.

this change aligns the behavior with run-all so that it will work with terragrunt cli options, properly display order based on apply and/or destory, and can be run within any folder that run-all could be executed in.

Fixes #2685
Fixes #2768

BREAKING CHANGE: technically this does change the behavior, in that it no longer requires terraform init. but i believe that more accurately reflects the desired behavior.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs -- based on conventions with sub commands, no additional documentation seemed required.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

Updated `output-module-groups` to work like run-all from any sub folder within a terragrunt configuration
Updated `output-module-groups` no longer requires `init` which greatly improves performance
Added support for terragrunt cli options for `output-module-groups`
Added support for `destroy` sub command for `output-module-groups` to generate reversed dependency ordered stack list

### Migration Guide

If any manual or automated process relied on the modules being initialized with `init` when running `output-module-groups` those processes will need to be updated to `init` via some other mechanism such as `terragrunt run-all init`

### CODE REVIEW

**Comment by bt-macole** [comment]
just noticed d5204846b35e4c8f45e51bbf7541c1430e70a7a4 has the same change to cli/commands/output-module-groups/action.go, happy to cherry-pick that here or visa versa.

**Comment by evsl** [comment]
> just noticed #2769 has the same change to cli/commands/output-module-groups/action.go ([d520484](https://github.com/gruntwork-io/terragrunt/commit/d5204846b35e4c8f45e51bbf7541c1430e70a7a4)), happy to cherry-pick that here or visa versa.

That would be neat! I was just poking about to consider updating my PR based on those changes, and I would certainly imagine that @denis256 would also like an integration test around the new functionality.

Additionally, relating to tests, my PR does fix the existing integration test for `output-module-groups` (it already existed, but it was erroneously passing due to the presence of a terragrunt.hcl file in the root of the test fixture, which I simply moved).

File: cli/commands/output-module-groups/command.go [hunk 1]
```diff
@@ -6,13 +6,42 @@ import (
...
+}
+
+func destroySubCommandFunc(opts *options.TerragruntOptions) *cli.Command {
+	return subCommandFunc(SubCommandDestroy, opts)
+}
+
+func subCommandFunc(cmd string, opts *options.TerragruntOptions) *cli.Command {
+	opts.TerraformCommand = cmd
+	return &cli.Command{
+		Name:   cmd,
```
Comment (denis256) [inline cli/commands/output-module-groups/command.go:44]: will be helpful to define `Usage` to show how subcommands can be used

image

**Review by denis256 — COMMENTED** [review]
Hello,
will be helpful to cherry-pick all changes in one PR, which looks like was already applied for `cli/commands/output-module-groups/action.go`, remaining only to remove `terragrunt.hcl`

PR misses the documentation page to show examples of how to use `output-module-groups` subcommands

File: cli/commands/output-module-groups/command.go (hunk 1 above)
Comment (bt-macole) [reply cli/commands/output-module-groups/command.go:45]: I originally had usage for the sub-commands but felt it was repetitive. updated!

**Comment by bt-macole** [comment]
> Hello,
> will be helpful to cherry-pick all changes in one PR, which looks like was already applied for cli/commands/output-module-groups/action.go, remaining only to remove terragrunt.hcl
>
> PR misses the documentation page to show examples of how to use output-module-groups subcommands

updated. also added table driven test to cover all subcommands

**Review by denis256 — APPROVED** [review]

**Comment by bt-macole** [comment]
@evsl good stuff!

thanks @denis256 🎉

**Pull request merged**